### PR TITLE
Added cmdline support to Open API add-on

### PR DESCRIPTION
**Pull request by psiinon**

### CODE REVIEW

File: src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java [hunk 1]
```diff
@@ -240,5 +249,59 @@ public URL getURL() {
...
+                "-openapifile <path>      " + Constant.messages.getString("openapi.cmdline.file.help"));
+        arguments[ARG_IMPORT_URL_IDX] = new CommandLineArgument("-openapiurl", 1, null, "", 
+                "-openapiurl <url>        " + Constant.messages.getString("openapi.cmdline.url.help"));
+        return arguments;
+    }
+
+    @Override
+    public void execute(CommandLineArgument[] args) {
+        if (arguments[ARG_IMPORT_FILE_IDX].isEnabled()) {
+            for (CommandLineArgument arg : args) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java]: This for loop should be removed (`args` contains both arguments, import file and URL).

File: src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java [hunk 2]
```diff
@@ -240,5 +249,59 @@ public URL getURL() {
...
+                Vector<String> params = arg.getArguments();
+                if (params != null) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java]: This if can be removed (not needed per previous loop change and `getArguments()` does no return null).

File: src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java [hunk 3]
```diff
@@ -240,5 +249,59 @@ public URL getURL() {
...
+                    for (String file : params) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java]: This for loop should be `for (String file : args[ARG_IMPORT_FILE_IDX].getArguments()) {` instead, to just use the arguments of import file.

File: src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java [hunk 4]
```diff
@@ -240,5 +249,59 @@ public URL getURL() {
...
+                        File f = new File(file);
+                        if (f.canRead()) {
+                            this.importOpenApiDefinition(f);
+                        } else {
+                            LOG.error("Cannot read Open API file: " + f.getAbsolutePath());
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java]: Should use `CommandLine.error(String)`.

File: src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java [hunk 5]
```diff
@@ -240,5 +249,59 @@ public URL getURL() {
...
+                        }
+                    }
+                }
+            }
+        } else if (arguments[ARG_IMPORT_URL_IDX].isEnabled()) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java]: Do these command line arguments need to be mutually exclusive?

File: src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java [hunk 6]
```diff
@@ -240,5 +249,59 @@ public URL getURL() {
...
+            for (CommandLineArgument arg : args) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java]: Same comments as above (regarding this loop and following statements).

File: src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java [hunk 7]
```diff
@@ -240,5 +249,59 @@ public URL getURL() {
...
+                Vector<String> params = arg.getArguments();
+                if (params != null) {
+                    for (String urlstr : params) {
+                        try {
+                            URI url = new URI(urlstr, false);
+                            this.importOpenApiDefinition(url);
+                        } catch (Exception e) {
+                            LOG.error(e.getMessage(), e);
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java]: Should use `CommandLine.error(String, Throwable)`.

File: src/org/zaproxy/zap/extension/openapi/ZapAddOn.xml [hunk 8]
```diff
@@ -5,7 +5,12 @@
 	<description>Imports and spiders Open API definitions.</description>
 	<author>ZAP Core Team plus Joanna Bona, Artur Grzesica, Michal Materniak and Marcin Spiewak</author>
 	<url></url>
-	<changes>Configure Swagger library logging.</changes>
+	<changes>
+	<![CDATA[
+		Configure Swagger library logging.<br>
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/openapi/ZapAddOn.xml]: This change can be removed and the version bumped :)

**Comment by psiinon** [comment]
Ha, looks like I ripped off the wrong code (or ripped the right code off badly;)
Hopefully fixed now.

**Comment by thc202** [comment]
Thanks!

**Pull request merged**